File: pratico.ts
class Arquivo {
    private nome: string;
    private tipo: string;
    private tamanho: number;

    constructor(nome: string, tipo: string, tamanho: number) {
        this.nome = nome;
        this.tipo = tipo;
        this.tamanho = tamanho;
    }

    public getTamanho(): number {
        console.log("Nome do arquivo: " + this.nome);
        console.log("Tipo do arquivo: " + this.tipo);
        return this.tamanho;
    }
}

class Pasta {
    private nome: string;
    private subpastas: Pasta[];
    private arquivos: Arquivo[];

    constructor(nome: string) {
        this.nome = nome;
        this.subpastas = [];
        this.arquivos = [];
    }

    public criarPasta(nome: string): void {
        const pasta: Pasta = new Pasta(nome);
        this.subpastas.push(pasta);
    }

    public criarArquivo(nome: string, tipo: string, tamanho: number): void {
        const arquivo: Arquivo = new Arquivo(nome, tipo, tamanho);
        this.arquivos.push(arquivo);
    }

    public calcularTamanho(): number {
        let tamanhoTotal: number = 0;
        for (const subpasta of this.subpastas) {
            tamanhoTotal += subpasta.calcularTamanho();
        }
        for (const arquivo of this.arquivos) {
            tamanhoTotal += arquivo.getTamanho();
        }
        return tamanhoTotal;
    }

    public excluir(): void {
        for (const subpasta of this.subpastas) {
            subpasta.excluir();
        }
        this.subpastas = [];
        this.arquivos = [];
    }

    public getSubpasta(nome: string): Pasta | undefined {
        for (const subpasta of this.subpastas) {
            if (subpasta.nome === nome) {
                return subpasta;
            }
        }
        return undefined;
    }
}

class Disco {
    private nome: string;
    private raiz: Pasta;

    constructor(nome: string) {
        this.nome = nome;
        this.raiz = new Pasta("Raiz");
    }

    public getNome(): string {
        return this.nome;
    }

    public criarPasta(nome: string): void {
        this.raiz.criarPasta(nome);
    }

    public criarArquivo(nome: string, tipo: string, tamanho: number): void {
        this.raiz.criarArquivo(nome, tipo, tamanho);
    }

    public calcularTamanho(): number {
        return this.raiz.calcularTamanho();
    }

    public excluir(): void {
        this.raiz.excluir();
    }

    public getRaiz(): Pasta {
        return this.raiz;
    }
}

const gdrive: Disco = new Disco("GDrive");

gdrive.criarPasta("Documents");
gdrive.criarPasta("Pictures");

const documents: Pasta | undefined = gdrive.getRaiz().getSubpasta("Documents");
if (documents) {
    documents.criarPasta("Work");
    documents.criarPasta("Personal");

    documents.criarArquivo("Report.docx", "Documento", 1024);
    documents.criarArquivo("Image.jpg", "Imagem", 2048);

    const tamanhoDocuments: number = documents.calcularTamanho();
    console.log("Tamanho total da pasta Documents: " + tamanhoDocuments + " bytes");

    documents.excluir();
} else {
    console.log("Pasta Documents não encontrada.");
}

File: pratico.java
import java.util.ArrayList;
import java.util.List;

class Arquivo {
    private String nome;
    private String tipo;
    private long tamanho;

    public Arquivo(String nome, String tipo, long tamanho) {
        this.nome = nome;
        this.tipo = tipo;
        this.tamanho = tamanho;
    }

    public long getTamanho() {
        return tamanho;
    }
}

class Pasta {
    private String nome;
    private List<Pasta> subpastas;
    private List<Arquivo> arquivos;

    public Pasta(String nome) {
        this.nome = nome;
        this.subpastas = new ArrayList<>();
        this.arquivos = new ArrayList<>();
    }

    public void criarPasta(String nome) {
        Pasta pasta = new Pasta(nome);
        subpastas.add(pasta);
    }

    public void criarArquivo(String nome, String tipo, long tamanho) {
        Arquivo arquivo = new Arquivo(nome, tipo, tamanho);
        arquivos.add(arquivo);
    }

    public long calcularTamanho() {
        long tamanhoTotal = 0;
        for (Pasta subpasta : subpastas) {
            tamanhoTotal += subpasta.calcularTamanho();
        }
        for (Arquivo arquivo : arquivos) {
            tamanhoTotal += arquivo.getTamanho();
        }
        return tamanhoTotal;
    }

    public void excluir() {
        for (Pasta subpasta : subpastas) {
            subpasta.excluir();
        }
        subpastas.clear();
        arquivos.clear();
    }
}

class Disco {
    private String nome;
    private Pasta raiz;

    public Disco(String nome) {
        this.nome = nome;
        this.raiz = new Pasta("Raiz");
    }

    public void criarPasta(String nome) {
        raiz.criarPasta(nome);
    }

    public void criarArquivo(String nome, String tipo, long tamanho) {
        raiz.criarArquivo(nome, tipo, tamanho);
    }

    public long calcularTamanho() {
        return raiz.calcularTamanho();
    }

    public void excluir() {
        raiz.excluir();
    }
}

public class Main {
    public static void main(String[] args) {
        Disco gdrive = new Disco("GDrive");

        // Criar pastas
        gdrive.criarPasta("Documents");
        gdrive.criarPasta("Pictures");

        // Criar subpastas
        Pasta documents = gdrive.getRaiz().getSubpasta("Documents");
        documents.criarPasta("Work");
        documents.criarPasta("Personal");

        // Criar arquivos
        documents.criarArquivo("Report.docx", "Documento", 1024);
        documents.criarArquivo("Image.jpg", "Imagem", 2048);

        // Calcular tamanho total da pasta "Documents"
        long tamanhoDocuments = documents.calcularTamanho();
        System.out.println("Tamanho total da pasta Documents: " + tamanhoDocuments + " bytes");

        // Excluir a pasta "Documents" e todos os seus arquivos e subpastas
        documents.excluir();
    }
}
